lib/index.js

```diff
@@ -48,11 +49,21 @@ exports.register = function (server, options, next) {
 
     server.ext('onPreAuth', function (request, reply) {
 
+        // Don't add session for specific routes
+
+        var requestSettings = Hoek.applyToDefaults(settings, request.route.settings.plugins.yar || {});
```

Comment:
This is expensive to perform on every request. Take a look at how crumb does the same thing by compiling the settings and saving it into the config.